pkg/cloudprovider/gcp/block_store.go

```diff
@@ -73,6 +76,30 @@ func (b *blockStore) Init(config map[string]string) error {
 	return nil
 }
 
+func extractProjectFromCreds() (string, error) {
+	credsBytes, err := ioutil.ReadFile(os.Getenv("GOOGLE_APPLICATION_CREDENTIALS"))
+	if err != nil {
+		return "", errors.WithStack(err)
+	}
+
+	var data map[string]interface{}
```

Comment:
You can do this with an inline (or top level) struct:
```go
type credentials struct {
  ProjectID string `json:"project_id"`
}

var c credentials
if err := json.Unmarshal(credsBytes, &c); err != nil {
  // ...
}

if c.ProjectID == "" {
 // error
}

return c.ProjectID

Reply:
oh, yeah, nice - i'll do this